src/packet/sym_encrypted_integrity_protected_data.js

```diff
@@ -183,16 +184,23 @@ class SymEncryptedIntegrityProtectedDataPacket {
     if (isArrayStream(encrypted)) encrypted = await streamReadToEnd(encrypted);
 
     let packetbytes;
+    let grammarValidator;
     if (this.version === 2) {
       if (this.cipherAlgorithm !== sessionKeyAlgorithm) {
         // sanity check
         throw new Error('Unexpected session key algorithm');
       }
       packetbytes = await runAEAD(this, 'decrypt', key, encrypted);
+      grammarValidator = getMessageGrammarValidator({ delayReporting: false });
     } else {
       const { blockSize } = getCipherParams(sessionKeyAlgorithm);
       const decrypted = await cipherMode.cfb.decrypt(sessionKeyAlgorithm, key, encrypted, new Uint8Array(blockSize));
 
+      // Grammar validation cannot be run before message integrity has been enstablished,
+      // to avoid leaking info about the unauthenticated message structure.
+      const releaseUnauthenticatedStream = util.isStream(encrypted) && config.allowUnauthenticatedStream;
+      grammarValidator = getMessageGrammarValidator({ delayReporting: releaseUnauthenticatedStream });
```

Comment:
I don't think this condition is correct, because even if we're not streaming or `config.allowUnauthenticatedStream` is false, we should _still_ wait with reporting grammar errors until after the MDC check has been done. So I think this should just be:
```suggestion
      // Grammar validation cannot be run before message integrity has been enstablished,
      // to avoid leaking info about the unauthenticated message structure.
      grammarValidator = getMessageGrammarValidator({ delayReporting: true });
```

Reply:
If we are not streaming unauthenticated data, all the packet bytes are authenticated before being passed to the parser, because there is a readToEnd here: https://github.com/ProtonMail/openpgpjs/blob/main/src/packet/sym_encrypted_integrity_protected_data.js#L213; so the grammar checker can throw as soon as it detects an error.

Reply:
OK yes, fair enough.
But then still, since we have all the data available in that case, what's the advantage of checking after every packet? We're basically checking the grammar repeatedly after every packet, even though we already have all of them. So I think it's faster to just check it once at the end, especially in the common case of the message being valid.
(But, I guess that's less critical.)

Reply:
Because we can avoid parsing the rest of the stream if there is an issue. It's a cheap check

Reply:
Yes sure, but again, it doesn't really make sense to optimize for the "if there's an issue" case at the cost of the happy path

Reply:
It does if the cost for the happy path is zero. It's checking the content an array of numbers that has like 5 elements at most 😅

Reply:
I'm not entirely convinced that it's free, there's a bunch of allocations in there and the regression test complains. But anyway, we can refactor this later if needed.

Reply:
Yeah the test failed for an actual regression 👼 pushed a fix